src/librustdoc/clean/mod.rs

```diff
@@ -1610,7 +1610,6 @@ pub(crate) fn clean_middle_ty<'tcx>(
             type_: Box::new(clean_middle_ty(ty, cx, None)),
         },
         ty::FnDef(..) | ty::FnPtr(_) => {
-            let ty = cx.tcx.lift(this).expect("FnPtr lift failed");
```

Comment:
But technically this is the only fix needed -- this should've been `cx.tcx.lift(ty)`, but also isn't needed at all.